node/declaration.ts

```diff
@@ -146,6 +199,79 @@ export const parsePattern = (pattern: string) => {
     },
   })
 
+  // The `extractedExclusionPatterns` property works as a shut-off valve: if
+  // it's not supplied, don't even traverse the AST again to further process
+  // lookaheads.
+  if (extractedExclusionPatterns) {
+    const exclusionPatterns = [...lookaheads].map((lookahead) => getExclusionPatternFromLookahead(regexp, lookahead))
+
+    extractedExclusionPatterns.push(...exclusionPatterns)
```

Comment:
have you thought about introducing a logging-only mechanism that can give you data on real-world examples so you can spot check if this code is correct?